### ISSUE TITLE
Add user attributes / tags

### ISSUE THREAD

**Issue by ASankaran** [HackIllinois/api-user#15]
- [ ] Add the ability to tag users with arbitrary attributes

Experience team wants the ability to add classifications / attributes to users within a certain role. These attributes do not modify the user's permissions within the api. However they may be useful for displaying / sending specialized information to a subset of users.

**Comment by PradyumnaShome** [comment]
This is what I propose:
- This should be similar to the `class` / `className`(in JSX) attribute of HTML elements.
- Let's have a `Tags []string`, or even better, some kind of HashSet equivalent(to avoid duplicates) for each user.
- Functions:
1. `AddTag` - Appends a tag(string) to the list
2. `RemoveTag` - Removes the specified tag
3. `UpdateTag` - Replaces the list with the given list
4. Filter functions on a list of users that select a subset of users from a list - this could be integrated with the mailing list feature

This change will warrant an equivalent number of patches as were done for `FirstName` + `LastName`.

**Comment by ASankaran** [comment]
I agree we should add a `Tags []string` field to the user struct. However I'd say we don't need the `UpdateTags` endpoint. The tagging system should be set up to be fairly simple so an `AddTag` and `RemoveTag` endpoint should be sufficient.

For filtering we want a `GET /user/filter/?tags=TAG1,TAG2` endpoint which will return all users than contain either `TAG1`, `TAG2` or both.

I'm not sure why this requires more than 1 patch to establish the core tagging functionality. Everything related to tag management is done in the user service. A potential second patch could be to the mailing service to send email to all users with a certain tag. However experience hasn't asked for that feature yet. Plus that functionality can be done by using the filter by tag endpoint we will be adding with the add users to mailing list and send mail to mailing list endpoints already in the mail service.

**Comment by PradyumnaShome** [comment]
> Plus that functionality can be done by using the filter by tag endpoint we will be adding with the add users to mailing list and send mail to mailing list endpoints already in the mail service.

Sounds like a plan.
